models/genmodels.go

```diff
@@ -118,11 +120,20 @@ func printInputStruct(g *swagger.Generator, op *spec.Operation) error {
 }
 
 func printInputValidation(g *swagger.Generator, op *spec.Operation) error {
-	capOpID := swagger.Capitalize(op.ID)
-	g.Printf("// Validate returns an error if any of the %sInput parameters don't satisfy the\n",
-		capOpID)
-	g.Printf("// requirements from the swagger yml file.\n")
-	g.Printf("func (i %sInput) Validate() error{\n", capOpID)
+	sspp, paramName := swagger.SingleStringPathParameter(op)
+	if sspp {
+		capOpID := swagger.Capitalize(op.ID)
+		g.Printf("// Validate%sInput returns an error if the input parameter doesn't\n",
+			capOpID)
+		g.Printf("// satisfy the requirements in the swagger yml file.\n")
+		g.Printf("func Validate%sInput(%s string) error{\n", capOpID, paramName)
+	} else {
+		capOpID := swagger.Capitalize(op.ID)
+		g.Printf("// Validate returns an error if any of the %sInput parameters don't satisfy the\n",
+			capOpID)
+		g.Printf("// requirements from the swagger yml file.\n")
+		g.Printf("func (i %sInput) Validate() error{\n", capOpID)
```

Comment:
I think I prefer to keep it as-is (method on the type). As written I think the standalone / package-level `Validate*` method introduced here is not ideal... it's one of the downsides of not having the struct anymore.